Common/Securities/SecurityPortfolioManager.cs

```diff
@@ -596,6 +596,11 @@ public List<SubmitOrderRequest> ScanForMarginCall(out bool issueMarginCallWarnin
         /// <param name="dividend">The dividend to be applied</param>
         public void ApplyDividend(Dividend dividend)
         {
+            // check if symbol exists in Securities
+            if (!Securities.ContainsKey(dividend.Symbol))
+            {
+                return;
+            }
```

Comment:
We are double-checking, since the same if-statement is found in `AlgorithmManager`. 
Same for `ApplySplitTo Equities`.

Reply:
I have removed the double check both here and in ApplyToEqities, 
commit c4f54610232a296fc5f0985f7b23a75b443492a6